sympy/core/basic.py

```diff
@@ -842,6 +843,10 @@ def subs(self, *args, **kwargs):
                    When a single argument is passed to subs
                    it should be a dictionary of old: new pairs or an iterable
                    of (old, new) tuples."""))
+        elif isinstance(args[1], Vector) and self.find(args[0]):
```

Comment:
I forget now how this is handled, but I think this should be handled by creating a pertinent `_eval_subs` routine in the relevant object. I think we had to do something like this with Integral (by not letting the dummy variable be targeted by substitution). Putting it there keeps the Basic routine from becoming cluttered with special cases.